Felix/remove kinect

FelixKHWong opened the pull request:
Removing the kinect from the movo and adding in the realsense for ROS and for RVIZ

FelixKHWong committed:
Change from kinect to realsense for perception launch

replace some kinect point topics

Replace kinect2 with d435 for srdf ignore collisions

Remove most kinect files

Add some realsense model files

Change d435_link to camera_link

Change kinect ptclouds to camera/depth/color/points

Replace kinect with d435 in xacro files

Replace run dependecy on iai_kinect with realsense-ros

Remove kinect collision stl

Remap default_pointing_frame to realsense camera_link

Replace kinect rviz file references with realsense

Remove kinect install dependency

Try to add variable to launch pointcloud

Fix urdf.xacro file issues

Adjust realsense position in urdf

Remove kinect collision file

FelixKHWong commented on movo_common/movo_description/urdf/movo_components/movo_pan_tilt.urdf.xacro:
```diff
@@ -136,9 +136,10 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-			<origin xyz="0.05825 -0.0105875 -0.026" rpy="${-M_PI/2} 0 ${-M_PI/2}" />
-		</xacro:kinect_one_sensor>
+		<xacro:sensor_d435 parent="tilt_link">
+			<origin xyz="0 0 0" rpy="${-M_PI/2} 0 ${-M_PI/2}" />
```
These numbers may need changing

FelixKHWong commented on movo_desktop/movo_viz/rviz/robot_.rviz:
```diff
@@ -634,7 +639,7 @@ Visualization Manager:
     - Class: rviz/Image
       Enabled: false
-      Image Topic: /kinect2/sd/image_color_rect
+      Image Topic: /camera/depth/image_rect_raw
```
might change this topic to be the color version but this name seemed closer to the previous

younesr1 commented:
have u run `roslaunch movo_viz view_robot.launch` to verify the camera shows up correctly? also what things have u done to try to validate this change? maybe fix movo2.launch first then run it and make sure these changes work as expected

FelixKHWong commented:
Sure I'll give that a shot

bdschnapp reviewed (approved):
looks good to me, only hesitation is it seems like there were significantly more files removed than added. I'm assuming this is just the kinect and realsesne being structured differently?

FelixKHWong commented:
I've added some changes so that the realsense is in the correct spot in the movo model and I've added some changes to be able to properly launch the rviz files. I solved the errrors with movo2.launch. The solution for the time sync issue is to add a printf with the password and pipe it to the sudo -S command. The password for my computer is not the robohub one so I won't be adding it but I'll put in the necessary change for the movo
![realsense position](https://user-images.githubusercontent.com/42899188/195134534-a905813d-a347-480e-9c56-f6ec12b001df.png)

FelixKHWong commented:
Please do check if the script issue with the password is feasible or not. I'm assuming since the password is that it is fine to put in in the script but we can remove it if needed

younesr1 commented on movo_desktop/movo_viz/launch/view_robot.launch:
```diff
@@ -3,6 +3,13 @@
 
+  <include file="$(find movo_description)/launch/movo_upload.launch"/>
+
+  <include file="$(find movo_moveit_config)/launch/move_group.launch">
```
its a little strange for the view_robot launch file to be launching move groups. isnt the point of this launch file to just view the robot?

FelixKHWong replied:
Yeah that's what I thought but when I try to run the view_robot.launch file it raises an error about not having the movo description and after I add the upload launch file it states that I need the semantic description. Did you not see these when you ran it?

younesr1 replied:
hmm why does move_group need to be launched with view_robot.launch? did u just try adding different launch files until it worked? i dont think move_group should be launched with view_robot

FelixKHWong replied:
![view_robot_err](https://user-images.githubusercontent.com/42899188/195730381-8745d996-8e83-4370-ab6e-25f7a5f386b4.png)
Yeah I tried to resolve the errors I found when running the launch file with roslaunch. I added the movo_upload to get the robot description and then the movo_group launch to fix a semantic description error. If this runs on master fine then perhaps I'm missing a change and I can remove the lines

younesr1 commented:
> I've added some changes so that the realsense is in the correct spot in the movo model and I've added some changes to be able to properly launch the rviz files. I solved the errrors with movo2.launch. The solution for the time sync issue is to add a printf with the password and pipe it to the sudo -S command. The password for my computer is not the robohub one so I won't be adding it but I'll put in the necessary change for the movo ![realsense position](https://user-images.githubusercontent.com/42899188/195134534-a905813d-a347-480e-9c56-f6ec12b001df.png)

I dont think i understand what you mean. you've run movo2.launch locally? can u collect a rosbag of it running and post it? also writing passwords in source code is not great imo. there should be a better way

FelixKHWong commented:
Yeah I was running movo2 launch locally. It doesn't detect the realsense so it probably doesn't do anything but it was raising the time sync error we saw before so I worked on fixing it. I can try to add the user to the list of sudoers but so far when I've tried it it hasn't worked

FelixKHWong commented:
Removed password from history

FelixKHWong commented:
[2022-10-11-13-20-35.zip](https://github.com/nightingale-project/kinova-movo/files/9758398/2022-10-11-13-20-35.zip)
Here's the rosbag file which I had to zip to put here. It seems liek tehres a movo group error and a python error regarding the planning scene interface. I'll take a look at those

FelixKHWong commented:
https://github.com/ros-industrial/universal_robot/issues/413 https://github.com/ros-planning/moveit_tutorials/issues/513. Seems like a move_group was deprecated and changed

FelixKHWong commented on movo_common/movo_third_party/moveit_python/src/moveit_python/planning_scene_interface.py:
```diff
@@ -25,7 +25,7 @@
 # POSSIBILITY OF SUCH DAMAGE.
 
-import thread, copy
+import _thread, copy
```
library updated for python3

younesr1 replied:
is this right? have u seen this?
https://stackoverflow.com/questions/5568555/thread-vs-threading
"_thread is infrastructure code that is used to implement threading, and normal Python code shouldn't be going anywhere near it."
"threading adds some additional automatic accounting, as well as several convenience utilities, all of which makes it the preferred option for standard Python code."

FelixKHWong replied:
Good catch I didn't know about that and I was hesitant to change the code but after reading more of the code the thread library is only used to create a mutex and acquire or release so I think I've correctly replaced the functionality with the new threading library(coming in the next push)

FelixKHWong replied:
https://peps.python.org/pep-3108/#obsolete

FelixKHWong commented on movo_moveit_config/launch/move_group.launch:
```diff
@@ -51,7 +51,7 @@
     <!-- MoveGroup capabilities to load -->
     <param name="capabilities" value="move_group/MoveGroupCartesianPathService
-				      move_group/MoveGroupExecuteService
+				      move_group/MoveGroupExecuteTrajectoryAction
```
Old group was deprecated and replaced with new group

younesr1 commented:
do we still want `./movo_common/movo_description/meshes/sensors/collision/kinect_one_collision.stl`? its not used?

younesr1 commented on movo_robot/movo_bringup/launch/perception/movo_perception.launch:
```diff
@@ -1,112 +1,150 @@
+  
+
+  <group ns="$(arg camera)">
+    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
```
does this launch the nodelet manager for u?

FelixKHWong replied:
![managernode](https://user-images.githubusercontent.com/42899188/195731590-7fda7288-0812-442b-86aa-5cc40e2144a7.png)
![nodes](https://user-images.githubusercontent.com/42899188/195731591-5e2cb413-7d9b-4701-9512-cca6e5867e5c.png)
 I think it's opening a manager for the camera. I'm not that familiar with nodes so I'm not sure if there's supposed to be a node for everything or something

FelixKHWong commented:
> do we still want `./movo_common/movo_description/meshes/sensors/collision/kinect_one_collisi`

I thought I deleted it before but I missed it. I'll remove it now

FelixKHWong commented:
![Screenshot from 2022-10-15 13-36-46](https://user-images.githubusercontent.com/42899188/196000431-477b2124-a3f3-43ba-9ef2-a57f989a7739.png)
I finally figured out how to sudo with no password for the ros file. I'll have to edit the sudoers file on movo so I'll do that next time I'm in

FelixKHWong commented:
> looks good to me, only hesitation is it seems like there were significantly more files removed than added. I'm assuming this is just the kinect and realsesne being structured differently?

Yeah most of the kinect stuff is inside the movo repo but for realsense we include it as a ROS package so the code for it is in /opt

younesr1 reviewed (approved):
lgtm! good stuff felix!

younesr1 commented:
actually theres one more thing i think u should do
- [x] update the Camera section in the dev_notes.txt

FelixKHWong committed:
Update dev notes for realsense

younesr1 commented:
so to be clear, none of these changes will be ported the robohub fork of this repo, right? your 2 commit names imply this

FelixKHWong commented:
> so to be clear, none of these changes will be ported the robohub fork of this repo, right? your 2 commit names imply this

Yeah this is only intended for those who want to use a realsense with the movo. This commit should not be taken for those who are using the default movo setup
